ts/private/ts_project_worker.js

```diff
@@ -72,6 +77,25 @@ function createEmitCacheAndDiagnosticsProgram(
         return emit(targetSourceFile, writeF, cancellationToken, emitOnlyDtsFiles, customTransformers);
     };
 
+    /** Lib Cache */
+    const getSourceFile = host.getSourceFile;
+    host.getSourceFile = (fileName) => {
+        if (libCache.has(fileName)) {
+            host.debuglog?.(`cache hit for default lib ${fileName}`)
+            return libCache.get(fileName);
+        }
+        const sf = getSourceFile(fileName);
+        if (sf && 
+            fileName.includes('external') && 
+            fileName.includes('typescript@') && 
+            fileName.includes('node_modules/typescript/lib')
```

Comment:
actually, it's not a single lib. this will include everything under `typescript/lib folder`.  more specifically this folder https://github.com/microsoft/TypeScript/tree/main/src/lib